### PR TITLE
MINOR: put registerChannel after connected succeeded

### PR DESCRIPTION
**Pull request by walkwalkwalk**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Can you please elaborate why the change is needed?

**Comment by walkwalkwalk** [comment]
Thanks for review. I think if socketChannel fail to  connect to InetSocketAddress, there is no need to execute  registerChannel.

File: clients/src/main/java/org/apache/kafka/common/network/Selector.java [hunk 1]
```diff
@@ -212,9 +212,9 @@ public void connect(String id, InetSocketAddress address, int sendBufferSize, in
         try {
             configureSocketChannel(socketChannel, sendBufferSize, receiveBufferSize);
             boolean connected = doConnect(socketChannel, address);
-            SelectionKey key = registerChannel(id, socketChannel, SelectionKey.OP_CONNECT);
 
             if (connected) {
+                SelectionKey key = registerChannel(id, socketChannel, SelectionKey.OP_CONNECT);
```
Comment (hachikuji) [inline clients/src/main/java/org/apache/kafka/common/network/Selector.java:217]: I don't think this works. This branch only handles connections which are completed "immediately" in `doConnect`. This is not the common case, which is why all of the test cases in `SelectorTest` fail with this change.

**Comment by walkwalkwalk** [comment]
@hachikuji, Thanks for the review. I will close this PR.

**Pull request closed without merging**